### ISSUE TITLE
SSH errors in prow e2e jobs

### ISSUE THREAD

**Issue by krzyzacy** [kubernetes/test-infra#2602]
```
ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
```

For example, https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-ubuntu/6#k8sio-ssh-should-ssh-to-all-nodes-and-run-commands, and https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-reboot-release-1-4/75?log#log

Maybe issues with the binded ssh key? Not sure yet.

cc @yguo0905 

/assign @krzyzacy @spxtr

**Comment by yguo0905** [comment]
These are the GCP ssh keys in the tests, which are set in prow/config.yaml.
```
JENKINS_GCE_SSH_PRIVATE_KEY_FILE=/etc/ssh-key-secret/ssh-private
JENKINS_GCE_SSH_PUBLIC_KEY_FILE=/etc/ssh-key-secret/ssh-public
```

Here are the AWS keys, which are not defined in prow/config.yaml.
```
JENKINS_AWS_SSH_PRIVATE_KEY_FILE=/root/.ssh/kube_aws_rsa
JENKINS_AWS_SSH_PUBLIC_KEY_FILE=/root/.ssh/kube_aws_rsa.pub
```

Not sure what's in /etc/ssh-key-secret/ssh-private. Maybe we want to unset them in prow/config.yaml?

**Comment by krzyzacy** [comment]
You do not want to include AWS keys in gce jobs though.

I think we are missing something else? The [gce canary job](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-prow-canary/15609) uses the SSH key and seems still green.

**Comment by yguo0905** [comment]
> You do not want to include AWS keys in gce jobs though.

I didn't set the AWS keys, they are set by default. That's why I guess not setting the GCP keys would make a difference.

> I think we are missing something else? The gce canary job uses the SSH key and seems still green.

Maybe the job does not use any SSH operations? But I do see the following errors.
```
I0428 18:59:31.242] ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return code [255].
```

**Comment by yguo0905** [comment]
@fejta

**Comment by yguo0905** [comment]
Also, is there an example on triggering a periodic job from prow locally, so that the prow configurations can be tested? I'd like to test new jobs before submitting PRs. (BTW, running test-infra/bootstrap.py for these jobs locally did not have the SSH error.)

I've looked at https://github.com/kubernetes/test-infra/tree/master/prow#how-to-test-prow and https://github.com/kubernetes/test-infra/tree/master/prow/cmd/phony/examples, but didn't find an easy way to do so.

**Comment by fejta** [comment]
Looks like the ssh creds are mounted to /etc/ssh-key-secret
In order for SSH to work you need the creds to be somewhere ssh will find them, such as $HOME/.ssh/

See: https://github.com/kubernetes/test-infra/blob/master/scenarios/kubernetes_e2e.py#L102

**Comment by fejta** [comment]
It looks like the key is rejected:
```
W0429 07:40:53.191] Permission denied (publickey).
```

However the test logs show that the public key exists and is getting uploaded to project metadata:
```
W0429 07:16:58.386] 2017/04/29 07:16:58 main.go:377: Checking existing of GCP ssh keys...
W0429 07:16:58.386] 2017/04/29 07:16:58 main.go:387: Checking presence of public key in k8s-jkns-e2e-gce-ubuntu
W0429 07:16:58.386] 2017/04/29 07:16:58 util.go:156: Running: gcloud compute --project=k8s-jkns-e2e-gce-ubuntu project-info describe
W0429 07:16:59.086] 2017/04/29 07:16:59 util.go:158: Step 'gcloud compute --project=k8s-jkns-e2e-gce-ubuntu project-info describe' finished in 699.763726ms
W0429 07:16:59.086] 2017/04/29 07:16:59 main.go:393: Uploading public ssh key to project metadata...
W0429 07:16:59.086] 2017/04/29 07:16:59 util.go:98: Running: gcloud compute --project=k8s-jkns-e2e-gce-ubuntu config-ssh
W0429 07:16:59.895] WARNING: No host aliases were added to your SSH configs because you do not have any instances. Try running this command again after creating some instances.
W0429 07:16:59.929] 2017/04/29 07:16:59 util.go:100: Step 'gcloud compute --project=k8s-jkns-e2e-gce-ubuntu config-ssh' finished in 842.915411ms
```

If you check the project metadata it looks like that key is getting associated with the user root, which the ubuntu image is rejecting ssh requests for:

```
I0429 07:43:28.336] Please login as the user "ubuntu" rather than the user "root".
```

So I suspect the issue is that we need to attempt to login as a user other than root

**Comment by fejta** [comment]
I suspect we do not have this in other jobs because we set USER=jenkins in the env:
```
I0429 00:03:05.605] USER=jenkins
```

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce/7577?log#log

**Comment by krzyzacy** [comment]
Fixed by #2618
